src/main/java/org/terasology/dynamicCities/buildings/BuildingManager.java

```diff
@@ -180,7 +180,7 @@ public void postBegin() {
         return Optional.empty();
     }
 
-    public Optional<GenericBuildingComponent> getRandomBuildingOfZoneForCulture(String zone, Rect2i shape, CultureComponent cultureComponent) {
+    public GenericBuildingComponent getRandomBuildingOfZoneForCulture(String zone, Rect2i shape, CultureComponent cultureComponent) {
```

Comment:
Why did you change this interface? I personally think that `Optional` conveys the intended interface way better than the Java's typical `null` pattern. I still find it counter-intuitive that a method `public X getFoo()` might return "nothing" as part of its interface.